lib/std/syncio.nim

```diff
@@ -376,7 +376,7 @@ when defined(nimdoc) or (defined(posix) and not defined(nimscript)) or defined(w
       flags = if inheritable: flags and not FD_CLOEXEC else: flags or FD_CLOEXEC
       result = c_fcntl(f, F_SETFD, flags) != -1
     else:
-      result = setHandleInformation(cast[IoHandle](f), HANDLE_FLAG_INHERIT,
+      result = setHandleInformation(cast[IoHandle](int(f)), HANDLE_FLAG_INHERIT,
```

Comment:
why is a conversion to int needed here? if f is smaller than iohandle, this would be a bug

Reply:
`f` is a `cint`, `IoHandle` is a `distinct pointer`.

Reply:
that sounds fishy - how can that work even? windows functions like createfile return a `Handle`, not a cint afair - either that handle is trucated or this function is being called with garbage

Reply:
```nim
type
  FileHandle* = cint ## type that represents an OS file handle; this is
                      ## useful for low-level file access
```

https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fdopen-wfdopen?view=msvc-170

The file descriptor of the open file is `cint` on Windows.

Reply:
no, that's the C emulation layer - files are Handle on windows: 

https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilea

To us other "native" functions, c file numbers must be converted first:
https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/get-osfhandle?source=recommendations&view=msvc-170

Reply:
https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/get-osfhandle?view=msvc-170


```nim
proc getOsFileHandle*(f: File): FileHandle =
  ## Returns the OS file handle of the file `f`. This is only useful for
  ## platform specific programming.
  when defined(windows):
    result = FileHandle getOsfhandle(cint getFileHandle(f))
  else:
    result = c_fileno(f)
```

It is truncated here.

Reply:
:scream:

Reply:
Well, we should deprecate it and fixes its use case in the stdlib.

Reply:
A bit breaking change => https://github.com/nim-lang/Nim/pull/21106 perhaps we need to add a new proc